### ISSUE TITLE
multiple t/uv.t failures with Perl 5.006000

### ISSUE THREAD

**Issue by atoomic** [Dual-Life/Devel-PPPort#72]
using a perl `v5.6.0` I can notice that the test t/uv.t has multiple failures
note that the test is passing with `v5.6.1` [you need changes from d635b13e74b2f793e64dc5a1ccd04df3e65c5e2e]

currently we have disabled the test for 5.006 to have a clean travis run
```
 BEGIN { if ("$]" le '5.006') { skip 'skip: broken utf8 support', 0 for 1..52; exit; } }
```

but if you disable that workaround, here is the full output
```
> perl -v

This is perl, v5.6.0 built for darwin
(with 1 registered patch, see perl -V for more detail)

Copyright 1987-2000, Larry Wall

Perl may be copied only under the terms of either the Artistic License or the
GNU General Public License, which may be found in the Perl 5.0 source kit.

Complete documentation for Perl, including FAQ lists, should be found on
this system using `man perl' or `perldoc perl'.  If you have access to the
Internet, point your browser at http://www.perl.com/, the Perl Home Page.
>  perl -Mblib t/uv.t
Using /Users/nicolas/workspace/perl-modules/Devel-PPPort/blib
1..52
ok 1
ok 2
ok 3
ok 4
ok 5
ok 6
ok 7
ok 8
ok 9
ok 10
ok 11
ok 12
ok 13
not ok 14
# Test 14 got: '0' (t/uv.t at line 69)
#    Expected: '65'
not ok 15
# Test 15 got: '-1' (t/uv.t at line 70)
#    Expected: '1'
ok 16
not ok 17
# Test 17 got: '-1' (t/uv.t at line 74)
#    Expected: '1'
not ok 18
# Test 18 got: '0' (t/uv.t at line 77)
#    Expected: '65'
not ok 19
# Test 19 got: '-1' (t/uv.t at line 78)
#    Expected: '1'
ok 20
not ok 21
# Test 21 got: '-1' (t/uv.t at line 82)
#    Expected: '1'
not ok 22
# Test 22 got: '0' (t/uv.t at line 89)
#    Expected: '256'
not ok 23
# Test 23 got: '-1' (t/uv.t at line 90)
#    Expected: '2'
ok 24
ok 25
not ok 26
# Test 26 got: '0' (t/uv.t at line 103)
#    Expected: '65533'
not ok 27
# Test 27 got: '-1' (t/uv.t at line 104)
#    Expected: '1'
ok 28
ok 29
not ok 30
# Test 30 got: '0' (t/uv.t at line 167)
#    Expected: '1' (utf8_to_uvchr_buf(""); '' contains '(?-xism:empty)')
not ok 31
# Test 31 got: '0' (t/uv.t at line 172)
#    Expected: '65533' (returned value utf8_to_uvchr_buf(""); warnings disabled)
not ok 32
# Test 32 got: '-1' (t/uv.t at line 173)
#    Expected: '0' (returned length utf8_to_uvchr_buf(""); warnings disabled)
ok 33
ok 34
not ok 35
# Test 35 got: '0' (t/uv.t at line 167 fail #2)
#    Expected: '1' (utf8_to_uvchr_buf("\xc4\xc5"); '' contains '(?-xism:non-continuation)')
not ok 36
# Test 36 got: '0' (t/uv.t at line 172 fail #2)
#    Expected: '65533' (returned value utf8_to_uvchr_buf("\xc4\xc5"); warnings disabled)
not ok 37
# Test 37 got: '-1' (t/uv.t at line 173 fail #2)
#    Expected: '1' (returned length utf8_to_uvchr_buf("\xc4\xc5"); warnings disabled)
ok 38
ok 39
not ok 40
# Test 40 got: '0' (t/uv.t at line 167 fail #3)
#    Expected: '1' (utf8_to_uvchr_buf("\xc4"); '' contains '(?-xism:short|1 byte, need 2)')
not ok 41
# Test 41 got: '0' (t/uv.t at line 172 fail #3)
#    Expected: '65533' (returned value utf8_to_uvchr_buf("\xc4"); warnings disabled)
not ok 42
# Test 42 got: '-1' (t/uv.t at line 173 fail #3)
#    Expected: '1' (returned length utf8_to_uvchr_buf("\xc4"); warnings disabled)
ok 43
ok 44
not ok 45
# Test 45 got: '0' (t/uv.t at line 167 fail #4)
#    Expected: '1' (utf8_to_uvchr_buf("\xc0\x81"); '' contains '(?-xism:overlong|2 bytes, need 1)')
not ok 46
# Test 46 got: '0' (t/uv.t at line 172 fail #4)
#    Expected: '65533' (returned value utf8_to_uvchr_buf("\xc0\x81"); warnings disabled)
not ok 47
# Test 47 got: '-1' (t/uv.t at line 173 fail #4)
#    Expected: '2' (returned length utf8_to_uvchr_buf("\xc0\x81"); warnings disabled)
ok 48
ok 49
not ok 50
# Test 50 got: '0' (t/uv.t at line 167 fail #5)
#    Expected: '1' (utf8_to_uvchr_buf("\xff\x80\x90\x90\x90\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf"); '' contains '(?-xism:Malformed UTF-8 character)')
not ok 51
# Test 51 got: '0' (t/uv.t at line 172 fail #5)
#    Expected: '65533' (returned value utf8_to_uvchr_buf("\xff\x80\x90\x90\x90\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf"); warnings disabled)
not ok 52
# Test 52 got: '-1' (t/uv.t at line 173 fail #5)
#    Expected: '13' (returned length utf8_to_uvchr_buf("\xff\x80\x90\x90\x90\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf"); warnings disabled)
```

I've the feeling that we are using unsigned int instead of signed in some places

copy: @khwilliamson

**Comment by khwilliamson** [comment]
On 5/15/19 4:34 PM, Nicolas R. wrote:
> using a perl |v5.6.0| I can notice that the test t/uv.t has multiple 
> failures

Pali says that this function  in 5.6.0 doesn't do what I'm expecting it 
to.  I'm thinking we just put a minimum of 5.6.1 for 
utf8_to_uvchr_buf().  Camel 3 points out severe issues with Unicode in 5.6.0

**Comment by khwilliamson** [comment]
Fixed in 3.54
